### DISCUSSION TITLE
Without ssh-keygen and a Module Error: No module name "pynq"

### DISCUSSION THREAD

**Discussion by AXT2185** [Xilinx/finn#1070] [Q&A]
Hello, Sorry to bother everyone. I have some problem when i try to deploy the tfc_end2end_example to pynq-z2. But when I try to deploy it by ssh. I found that I cannot find the file 'ssh-keygen'. How I can solve it? 
![屏幕截图 2024-05-07 202512](https://github.com/Xilinx/finn/assets/85671658/ae7b8e92-4553-4480-8730-20a1992fb2db)
And I used to try upload the zip package about model by jupyter notebook. But when I executed it. It pushed a ModuleError. How can I solve it?
![屏幕截图 2024-05-07 081556](https://github.com/Xilinx/finn/assets/85671658/9fa4f06a-fead-4249-bd75-dc207532112c)

**Comment by gustandil** [comment]
Hi,

Same problem using Pynq 3.0.1 in Ultra96v2. Any suggestion?
So far I understand, the pynq environment is for Xilinx user.
As root, I made "source /etc/profile.d/pynq_venv.sh" obtaining same result

Regards, G

**Reply by Mengzou01** [reply]
Hi!
I got the same problem as you, can you help me solve it？
Hope it didn't bother you.
Sincerely, L

**Comment by Mengzou01** [comment]
Hi,
The ERROR may be caused because the Python path used by sudo is not the correct one that contains the PYNQ module.
So, use the command: 'sudo -E /usr/local/share/pynq-venv/bin/python3' instead of 'sudo python3'.
